pkg/test/environment.go

```diff
@@ -503,6 +507,25 @@ func (e *Environment) Context() context.Context {
 	return e.ctx
 }
 
+func (e *Environment) StartEmbeddedJetstream() (*nats.Conn, error) {
```

Comment:
Let's add this as an option to `Start()` which would replace the default subprocess jetstream server (you can also keep this function). We should also be able to obtain the port for the embedded server the same way as the external jetstream.

Additionally, try running all the tests using the embedded jetstream to see which ones work and which don't. This could potentially speed up some tests.

Reply:
Let's add that as a separate task.  I don't want to couple the implementation of this feature with additional test modifications.

Reply:
#933 created to track this

Reply:
I agree, adjusting existing tests can be a separate task. The first part of my comment is still relevant though.